Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandSwitchCharacter.java

```diff
@@ -54,61 +54,47 @@ public void run(Map<String, String> parameters) throws UserErrorMessage {
 			direction = Integer.parseInt(parameters.get("direction"));
 			
 			//This method will return all characters that we can switch to. Filters out zambies and dead characters right off the bat.
-			List<CachedEntity> characters = db.getAlphabetSortedValidCharactersByUser(user.getKey());
 			
 			switch(direction) {
 			
-			//UP in characterswitcher.
 			case 1:
+			case 2:
+				List<CachedEntity> characters = db.getAlphabetSortedValidCharactersByUser(user.getKey());
+				
 				for(int i = 0; i < characters.size(); i++) {
-					CachedEntity c = characters.get(i);
-					//here, we are searching for the currentcharacter, so we know its index within the sorted list.
-					if(!GameUtils.equals(c.getKey(), currentCharacter.getKey)) continue;					
+					if(!GameUtils.equals(characters.get(0).getKey(), currentCharacter.getKey())) continue;
 					
-					//if we're at the first element in the list
-					if(i == 0) {
-						targetCharacterId = characters.get(characters.size()).getId();
+					//Move UP in character list.
+					if(direction == 1) {
+						//if we're at the front of the list, loop back to the back.
+						if(i == 0) {
+							targetCharacterId = characters.get(characters.size()).getId();
+							break;
+						}
+						
+						targetCharacterId = characters.get(i-1).getId();
 						break;
 					}
 					
-					targetCharacterId = characters.get(i-1).getId();
-					break;
-				}
-				break;
-				
-			//DOWN in characterswitcher
-			case 2:
-				for(int i = 0; i < characters.size(); i++) {
-					CachedEntity c = characters.get(i);
-					//here, we are searching for the currentcharacter, so we know its index within the sorted list.
-					if(!GameUtils.equals(c.getKey(), currentCharacter.getKey)) continue;					
-					
-					//if we're at the last element in the list.
-					if(i == characters.size()) {
-						targetCharacterId = characters.get(0).getId();
+					//Move DOWN in character list.
+					if(direction == 2) {	
+						//if we're at the end of the list, loop back to the front.
+						if(i == characters.size()) {
+							targetCharacterId = characters.get(0).getId();
+							break;
+						}
+						
+						//otherwise, grab the next one.
+						targetCharacterId = characters.get(i+1).getId();
```

Comment:
Can likely simplify both steps: `int dir = direction == 1 ? -1 : 1;`, then use that var with above suggested change instead of `- 1` to `+ dir`.